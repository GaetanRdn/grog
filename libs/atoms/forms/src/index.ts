export * from './lib/input/input.directive';
export * from './lib/autocomplete/autocomplete.component';
export * from './lib/label/label.component';
export * from './lib/checkbox/checkbox.component';
export * from './lib/select/select.component';
