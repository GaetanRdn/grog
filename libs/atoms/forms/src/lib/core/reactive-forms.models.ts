export type OnChangeFn<ValueType> = (value: ValueType | null) => void;
export type OnTouchedFn = () => void;
