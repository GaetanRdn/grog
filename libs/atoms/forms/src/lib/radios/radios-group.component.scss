@use '../../../../../styles/src/lib/utils';

@mixin gro-radios($config) {
  $colors: map-get($config, colors);
  $colorKeys: map-keys($colors);

  gro-radios-group {
    @include utils.set-typography($config, input);
    display: flex;
    min-width: fit-content;
    flex: 1;
    accent-color: utils.get-deep-value($colors, primary, background-color);

    label {
      color: utils.get-deep-value($colors, primary, coloFr);

      &:focus-within {
        input[type='radio'] {
          color: utils.get-deep-value($colors, primary, background-color);
        }
      }

      input[type='radio']::before {
        box-shadow: inset 1em 1em utils.get-deep-value($colors, primary, background-color);
      }
    }

    &.gro-vertical {
      flex-flow: column;

      gro-radio + gro-radio {
        margin-top: 1em;
      }
    }

    &:not(.gro-vertical) {
      gro-radio + gro-radio {
        margin-left: 1em;
      }
    }

    @each $key in $colorKeys {
      &.gro-#{$key} {
        accent-color: utils.get-deep-value($colors, $key, background-color);

        label {
          color: utils.get-deep-value($colors, $key, color);

          &:focus-within {
            input[type='radio'] {
              color: utils.get-deep-value($colors, $key, background-color);
            }
          }

          input[type='radio']::before {
            box-shadow: inset 1em 1em utils.get-deep-value($colors, $key, background-color);
          }
        }
      }
    }

    gro-radio {
      cursor: pointer;

      label {
        display: grid;
        grid-template-columns: 1rem auto;
        gap: 0.5rem;

        input[type=radio] {
          appearance: none;
          background-color: #FFF;
          margin: 0;
          font: inherit;
          color: currentColor;
          width: 1.15em;
          height: 1.15em;
          border: 0.15em solid currentColor;
          border-radius: 50%;
          display: grid;
          place-content: center;
          transform: translateY(-0.075em);
        }

        input[type='radio']::before {
          content: '';
          width: 0.65em;
          height: 0.65em;
          border-radius: 50%;
          transform: scale(0);
          transition: 120ms transform ease-in-out;
        }

        input[type='radio']:checked::before {
          transform: scale(1);
        }
      }
    }
  }
}
