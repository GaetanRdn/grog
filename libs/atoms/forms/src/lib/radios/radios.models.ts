import { RadioComponent } from './radio.component';

export class CheckedChange<ValueType> {
  constructor(public readonly source: RadioComponent<ValueType>, public readonly isUserInteraction = false) {}
}
