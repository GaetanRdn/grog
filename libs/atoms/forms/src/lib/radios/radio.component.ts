import { ChangeDetectionStrategy, Component, EventEmitter, HostListener, Input, Output } from '@angular/core';
import { CoerceBoolean, Required } from '@grorg/decorators';
import { BooleanInput } from '@grorg/types';
import { CheckedChange } from './radios.models';
import { RadiosGroupComponent } from './radios-group.component';

@Component({
  selector: 'gro-radio',
  template: `<label>
    <input type="radio" [name]="name" [value]="value" [checked]="checked" [disabled]="disabled ?? null" />
    <ng-content></ng-content>
  </label>`,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class RadioComponent<ValueType> {
  static ngAcceptInputType_checked: BooleanInput;
  static ngAcceptInputType_disabled: BooleanInput;

  @Input()
  public name!: string;

  @Input()
  @Required()
  public value!: ValueType;

  @Output()
  public readonly checkedChange: EventEmitter<CheckedChange<ValueType>> = new EventEmitter<CheckedChange<ValueType>>();

  constructor(private readonly _parent: RadiosGroupComponent<ValueType>) {}

  private _disabled = false;

  get disabled(): boolean {
    return this._disabled || this._parent.disabled;
  }

  @Input()
  @CoerceBoolean()
  set disabled(disabled: boolean) {
    this._disabled = disabled;
  }

  private _checked = false;

  get checked(): boolean {
    return this._checked;
  }

  @Input()
  @CoerceBoolean()
  set checked(checked: boolean) {
    if (this._checked !== checked) {
      this._checked = checked;
      this.checkedChange.emit(new CheckedChange(this, false));
    }
  }

  @HostListener('click')
  public onClick(): void {
    if (!this.disabled) {
      this._checked = true;
      this.checkedChange.emit(new CheckedChange(this, true));
    }
  }
}
