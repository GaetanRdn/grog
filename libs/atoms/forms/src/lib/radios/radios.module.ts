import { NgModule } from '@angular/core';
import { RadioComponent } from './radio.component';
import { RadiosGroupComponent } from './radios-group.component';

@NgModule({
  declarations: [RadiosGroupComponent, RadioComponent],
  exports: [RadiosGroupComponent, RadioComponent],
})
export class RadiosModule {}
