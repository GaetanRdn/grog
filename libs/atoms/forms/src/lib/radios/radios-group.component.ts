import {
  AfterContentInit,
  ChangeDetectionStrategy,
  Component,
  ContentChildren,
  EventEmitter,
  forwardRef,
  Input,
  NgZone,
  OnInit,
  Optional,
  Output,
  QueryList,
  Self,
} from '@angular/core';
import { AutoUnsubscribe, CoerceBoolean, Required } from '@grorg/decorators';
import { BooleanInput, EqualsFn, Nullable } from '@grorg/types';
import { defer, merge, Observable, Subscription } from 'rxjs';
import { startWith, switchMap, take } from 'rxjs/operators';
import { RadioComponent } from './radio.component';
import { CheckedChange } from './radios.models';
import { ControlValueAccessor, NgControl } from '@angular/forms';
import { OnChangeFn, OnTouchedFn } from '../core/reactive-forms.models';

@Component({
  selector: 'gro-radios-group',
  host: {
    '[class.gro-vertical]': 'vertical || null',
    '[class.gro-disabled]': 'disabled || null',
  },
  template: ` <ng-content></ng-content>`,
  styleUrls: ['./radios-group.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
@AutoUnsubscribe()
export class RadiosGroupComponent<ValueType> implements AfterContentInit, OnInit, ControlValueAccessor {
  static ngAcceptInputType_vertical: BooleanInput;
  static ngAcceptInputType_disabled: BooleanInput;

  @Input()
  @Required()
  public name!: string;

  @Input()
  public value: Nullable<ValueType> = null;

  @Input()
  @CoerceBoolean()
  public vertical = false;

  @Output()
  public readonly valueChange: EventEmitter<ValueType> = new EventEmitter<ValueType>();

  @ContentChildren(forwardRef(() => RadioComponent), { descendants: true })
  public readonly radios!: QueryList<RadioComponent<ValueType>>;

  private readonly _checkedChange: Observable<CheckedChange<ValueType>> = defer(() => {
    if (this.radios) {
      return this.radios.changes.pipe(
        startWith(this.radios),
        switchMap(() => merge(...this.radios.map((radio: RadioComponent<ValueType>) => radio.checkedChange)))
      );
    }

    return this._ngZone.onStable.pipe(
      take(1),
      switchMap(() => this._checkedChange)
    );
  });
  private _subscriptions: Subscription[] = [];

  constructor(private readonly _ngZone: NgZone, @Self() @Optional() private readonly _ngControl?: NgControl) {
    if (_ngControl) {
      _ngControl.valueAccessor = this;
    }
  }

  private _disabled = false;

  get disabled(): boolean {
    return this._disabled || Boolean(this._ngControl?.disabled);
  }

  @Input()
  @CoerceBoolean()
  set disabled(disabled: boolean) {
    this._disabled = disabled;
  }

  public ngOnInit(): void {
    this._subscriptions[this._subscriptions.length] = this._checkedChange.subscribe(
      (change: CheckedChange<ValueType>) => {
        this.value = change.source.value;

        if (change.isUserInteraction) {
          this._onChanged(this.value);
          this._onTouched();
          this.valueChange.emit(this.value);
        }
      }
    );
  }

  @Input()
  public equalsFn: EqualsFn<ValueType> = (value1: Nullable<ValueType>, value2: Nullable<ValueType>) =>
    value1 === value2;

  public ngAfterContentInit(): void {
    this._subscriptions[this._subscriptions.length] = this.radios.changes
      .pipe(startWith(this.radios))
      .subscribe((radios: QueryList<RadioComponent<ValueType>>) => {
        this.setRadioNames(radios);

        if (this.value !== null) {
          const found: RadioComponent<ValueType> | undefined = radios
            .toArray()
            .find((radio: RadioComponent<ValueType>) => this.equalsFn(radio.value, this.value));

          if (found && !found.checked) {
            found.checked = true;
          }
        } else {
          const checked: RadioComponent<ValueType> | undefined = radios
            .toArray()
            .find((radio: RadioComponent<ValueType>) => radio.checked);

          if (checked) {
            Promise.resolve().then(() => {
              this.value = checked.value;
              this.valueChange.emit(this.value);
            });
          }
        }
      });
  }

  public registerOnChange(fn: OnChangeFn<ValueType>): void {
    this._onChanged = fn;
  }

  public registerOnTouched(fn: OnTouchedFn): void {
    this._onTouched = fn;
  }

  public writeValue(value: RadiosGroupComponent<ValueType>['value']): void {
    if (!this.equalsFn(this.value, value)) {
      this.value = value;
    }
  }

  private _onTouched: OnTouchedFn = () => ({});

  private _onChanged: OnChangeFn<ValueType> = () => ({});

  private setRadioNames(radios: QueryList<RadioComponent<ValueType>>): void {
    radios.toArray().forEach((radio: RadioComponent<ValueType>) => (radio.name = this.name));
  }
}
