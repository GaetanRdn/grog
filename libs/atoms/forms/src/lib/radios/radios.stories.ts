import { Nullable } from '@grorg/types';
import { action } from '@storybook/addon-actions';
import { Args, Meta, moduleMetadata, Story } from '@storybook/angular';
import { RadioComponent } from './radio.component';
import { RadiosGroupComponent } from './radios-group.component';
import { RadiosModule } from './radios.module';

export default {
  title: 'atoms/forms/radios',
  component: RadiosGroupComponent,
  subcomponents: { RadioComponent },
  decorators: [moduleMetadata({ imports: [RadiosModule] })],
} as Meta;

const template: Story = (args: Args) => ({
  props: {
    value: args['value'],
    valueChange: action('log'),
    vertical: Boolean(args['vertical']),
    class: args['class'] ? 'gro-' + args['class'] : null,
  },
  template: `<gro-radios-group name="yesOrNo" [value]="value" (valueChange)="valueChange($event)" [vertical]="vertical" class="{{class}}">
<gro-radio [value]="1">Yes</gro-radio>
<gro-radio [value]="0">No</gro-radio>
</gro-radios-group>`,
});

export const Default = template.bind({});
Default.args = {
  class: 'primary',
};
Default.argTypes = {
  class: {
    options: [
      'primary',
      'primary-light',
      'primary-dark',
      'accent',
      'accent-light',
      'accent-dark',
      'warn',
      'warn-light',
      'warn-dark',
    ],
    control: { type: 'select' },
  },
};

export const Vertical = template.bind({});
Vertical.args = {
  vertical: true,
};

export const WithGroupValue = template.bind({});
WithGroupValue.args = {
  value: 1,
};

const templateWithCheckedRadio: Story = (args: Args) => ({
  props: {
    value: args['value'],
    valueChange: action('log'),
  },
  template: `<gro-radios-group name="yesOrNo" [value]="value" (valueChange)="valueChange($event)">
<gro-radio [value]="1">Yes</gro-radio>
<gro-radio [value]="0" checked>No</gro-radio>
</gro-radios-group>`,
});
export const WithCheckedRadio = templateWithCheckedRadio.bind({});
WithCheckedRadio.args = {
  value: null,
};

const templateWithObjectValue: Story = (args: Args) => ({
  props: {
    value: args['value'],
    equalsFn: (val1: Nullable<{ id: number; name: string }>, val2: Nullable<{ id: number; name: string }>) =>
      val1?.id === val2?.id,
    valueChange: action('log'),
  },
  template: `<gro-radios-group name="yesOrNo" [value]="value" [equalsFn]="equalsFn" (valueChange)="valueChange($event)">
<gro-radio [value]="{ id: 1, name: 'yes' }">Yes</gro-radio>
<gro-radio [value]="{ id: 2, name: 'no' }">No</gro-radio>
</gro-radios-group>`,
});
export const WithObjectValue = templateWithObjectValue.bind({});
WithObjectValue.args = {
  value: { id: 1, name: 'yes' },
};
