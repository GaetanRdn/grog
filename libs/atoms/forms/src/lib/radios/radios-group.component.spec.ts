import { Component } from '@angular/core';
import { fakeAsync, TestBed, tick, waitForAsync } from '@angular/core/testing';
import { TemplateLookup } from '@grorg/tests';
import { EqualsFn, Nullable } from '@grorg/types';
import { RadiosGroupComponent } from './radios-group.component';
import { RadiosModule } from './radios.module';
import { FormControl, ReactiveFormsModule } from '@angular/forms';
import { RadioComponent } from './radio.component';

describe('RadiosGroupComponent', () => {
  beforeEach(
    waitForAsync(() => {
      TestBed.configureTestingModule({
        declarations: [HostComponent, HostWithoutEqualsFnComponent, HostReactiveFormComponent],
        imports: [RadiosModule, ReactiveFormsModule],
      }).compileComponents();
    })
  );

  describe('Complex Object', () => {
    let templateLookup: TemplateLookup<HostComponent>;

    beforeEach(() => {
      templateLookup = new TemplateLookup(TestBed.createComponent(HostComponent));
      jest.spyOn(templateLookup.hostComponent, 'valueChange');
    });

    test('should create', () => {
      templateLookup.detectChanges();

      expect(templateLookup.firstChildElement).toMatchSnapshot();
    });

    test('should create with initial value', () => {
      templateLookup.hostComponent.value = { id: 2, name: 'no' };

      templateLookup.detectChanges();

      expect(templateLookup.firstChildElement).toMatchSnapshot();
      expect(templateLookup.queryAll<HTMLInputElement>('input')[1].checked).toBe(true);
    });

    test('should set value from checked', fakeAsync(() => {
      templateLookup.hostComponent.checked = 1;

      templateLookup.detectChanges();
      tick(100);

      expect(templateLookup.firstChildElement).toMatchSnapshot();
      expect(templateLookup.getComponent(RadiosGroupComponent).value).toEqual({ id: 1, name: 'yes' });
      expect(templateLookup.hostComponent.valueChange).toBeCalledWith({ id: 1, name: 'yes' });
    }));

    test('should set value when check', fakeAsync(() => {
      templateLookup.detectChanges();

      templateLookup.queryAll<HTMLInputElement>('input')[0].click();
      templateLookup.detectChanges();
      tick(100);

      expect(templateLookup.firstChildElement).toMatchSnapshot();
      expect(templateLookup.getComponent(RadiosGroupComponent).value).toEqual({ id: 1, name: 'yes' });
      expect(templateLookup.hostComponent.valueChange).toBeCalledWith({ id: 1, name: 'yes' });
    }));

    test('should create disabled', fakeAsync(() => {
      // WHEN
      templateLookup.hostComponent.disabled = true;
      tick(50);
      templateLookup.detectChanges();

      // THEN
      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }));
  });

  describe('Without EqualsFn', () => {
    let templateLookup: TemplateLookup<HostWithoutEqualsFnComponent>;

    beforeEach(() => {
      templateLookup = new TemplateLookup(TestBed.createComponent(HostWithoutEqualsFnComponent));
      templateLookup.detectChanges();
    });

    test('should create', fakeAsync(() => {
      // WHEN
      templateLookup.queryAll<HTMLInputElement>('input')[0].click();
      templateLookup.detectChanges();
      tick(100);

      // THEN
      expect(templateLookup.firstChildElement).toMatchSnapshot();
      expect(
        templateLookup
          .getComponent(RadiosGroupComponent)
          .equalsFn(templateLookup.getComponent(RadiosGroupComponent).value, 1)
      ).toBe(true);
    }));
  });

  describe('Reactive form', () => {
    let templateLookup: TemplateLookup<HostReactiveFormComponent>;

    beforeEach(() => {
      templateLookup = new TemplateLookup(TestBed.createComponent(HostReactiveFormComponent));
    });

    test('should create', fakeAsync(() => {
      templateLookup.detectChanges();
      expect(templateLookup.firstChildElement).toMatchSnapshot();
    }));

    test('should create with value', fakeAsync(() => {
      // WHEN
      templateLookup.hostComponent.control.setValue(true);
      templateLookup.detectChanges();

      // THEN
      expect(templateLookup.firstChildElement).toMatchSnapshot();
      expect(templateLookup.getComponent(RadioComponent).checked).toBe(true);
    }));

    test('should create disabled', fakeAsync(() => {
      // WHEN
      templateLookup.hostComponent.control.disable();
      templateLookup.detectChanges();

      // THEN
      expect(templateLookup.firstChildElement).toMatchSnapshot();
      expect(templateLookup.getComponent(RadiosGroupComponent).disabled).toBe(true);
    }));

    test('should change control value on click', fakeAsync(() => {
      // GIVEN
      templateLookup.hostComponent.control.setValue(false);
      templateLookup.detectChanges();

      // WHEN
      templateLookup.queryAll<HTMLInputElement>('input')[0].click();
      tick(100);
      templateLookup.detectChanges();

      // THEN
      expect(templateLookup.firstChildElement).toMatchSnapshot();
      expect(templateLookup.hostComponent.control.value).toBe(true);
    }));
  });
});

@Component({
  template: ` <gro-radios-group
    name="yesOrNo"
    [value]="value"
    [disabled]="disabled"
    [equalsFn]="equalsFn"
    (valueChange)="valueChange($event)"
  >
    <gro-radio [value]="{ id: 1, name: 'yes' }" [checked]="checked === 1">Yes</gro-radio>
    <gro-radio [value]="{ id: 2, name: 'no' }" [checked]="checked === 2">No</gro-radio>
  </gro-radios-group>`,
})
class HostComponent {
  public checked: Nullable<number> = null;

  public value: Nullable<{ id: number; name: string }> = null;

  public disabled = false;

  public equalsFn: EqualsFn<{ id: number; name: string }> = (
    val1: Nullable<{ id: number; name: string }>,
    val2: Nullable<{ id: number; name: string }>
  ) => val1?.id === val2?.id;

  public valueChange(val: { id: number; name: string }): void {
    this.value = val;
  }
}

@Component({
  template: ` <gro-radios-group name="yesOrNo">
    <gro-radio [value]="1">Yes</gro-radio>
    <gro-radio [value]="2">No</gro-radio>
  </gro-radios-group>`,
})
class HostWithoutEqualsFnComponent {}

@Component({
  template: ` <gro-radios-group name="trueOrFalse" [formControl]="control">
    <gro-radio [value]="true">True</gro-radio>
    <gro-radio [value]="false">False</gro-radio>
  </gro-radios-group>`,
})
class HostReactiveFormComponent {
  public readonly control: FormControl = new FormControl();
}
