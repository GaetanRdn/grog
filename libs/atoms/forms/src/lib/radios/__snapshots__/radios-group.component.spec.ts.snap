// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadiosGroupComponent Basic should create 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="1"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="2"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;

exports[`RadiosGroupComponent Complex Object should create 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;

exports[`RadiosGroupComponent Complex Object should create with initial value 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;

exports[`RadiosGroupComponent Complex Object should set value from checked 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;

exports[`RadiosGroupComponent Complex Object should set value when check 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;

exports[`RadiosGroupComponent Without EqualsFn should create 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="1"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="2"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;

exports[`RadiosGroupComponent should create 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;

exports[`RadiosGroupComponent should create with initial value 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;

exports[`RadiosGroupComponent should set value from checked 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;

exports[`RadiosGroupComponent should set value when check 1`] = `
<gro-radios-group
  name="yesOrNo"
>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      Yes
    </label>
  </gro-radio>
  <gro-radio>
    <label>
      <input
        name="yesOrNo"
        type="radio"
        value="[object Object]"
      />
      No
    </label>
  </gro-radio>
</gro-radios-group>
`;
